src/Extensibility.Kubernetes/KubernetesProvider.cs

```diff
@@ -443,6 +439,22 @@ public async Task<SaveResponse> Save(SaveRequest request, CancellationToken canc
                     throw new InvalidOperationException($"operation failed with status {ex.Response.StatusCode} and content: \n{ex.Response.Content}");
                 }
             }
+        }, error => new() { Error = error });
+
+        private async Task<T> DoWithUnhandledErrorHandling<T>(Func<Task<T>> executeFunc, Func<ExtensibilityErrorContainer, T> buildResponse)
+        {
+            try {
+                return await executeFunc();
+            }
+            catch (Exception ex)
+            {
+                return buildResponse(new ()
+                {
+                    Errors = new [] {
+                        new ExtensibilityError { Message = ex.Message, },
+                    },
+                });
+            }
         }
```

Comment:
Should this live in the extensibility host instead of in individual providers? It looks like the storage provider is not wrapping errors into response.